File: PHP/sizeify.php
<?php
function sizeify($url,$resizeto,$endpoint='http://sizeifyb.sjc.io') {
	$error_image = $endpoint.'/error?code=400';
	if (filter_var($url, FILTER_VALIDATE_URL, FILTER_FLAG_HOST_REQUIRED) === false) {
		return $error_image;
	}
	if (!preg_match('/[whpgbslc]\d{1,4}|\d{1,4}x\d{1,4}|original|big/', $resizeto)) {
		return $error_image;
	}
	$u = parse_url($url);
	$p = substr($u['path'],1);
	if (!empty($u['query'])) {
		$p .= '?' . $u['query'];
	}
	$r = [
		$endpoint,
		$u['scheme'],
		implode('.',array_reverse(explode('.',$u['host']))),
		$resizeto,
		rawurlencode($p)
	];
	return implode('/',$r);
}

File: python/sizeify.py
try:
	from urllib.parse import urlparse
	from urllib.parse import quote
	from urllib.request import pathname2url
except ImportError:
	from urllib import urlparse
	from urllib import quote
	from urllib import pathname2url
import re

endpoint = 'http://sizeifyb.sjc.io'

def badURL():
	return endpoint + '/error?code=400' 

def urlIsValid(u):
	e = urlparse(endpoint)
	isParadoxical = (e.netloc == u.netloc)
	isWellFormed = (u.scheme is not None and u.netloc is not None and u.path is not None)
	return (isWellFormed and not isParadoxical)

def resizeStringIsValid(resizeto):
	pat = re.compile("^([whslpbgc]\\d{1,4}|\\d{1,4}x\\d{1,4}|original|big)$")
	return ( pat.match(resizeto) is not None )

def url(sourceurl,resizeto):
	u = urlparse(sourceurl)
	safe_path = pathname2url(u.path.lstrip('/'));
	if not urlIsValid(u) or not resizeStringIsValid(resizeto):
		return badURL()
	else:
		reversehost = ".".join(reversed(u.netloc.split('.')))
		encodedpath = quote(safe_path,"")
		s = [ endpoint, u.scheme, reversehost, resizeto, encodedpath ]
		return "/".join(s)
